### DISCUSSION TITLE
How to Disable Collision Checking for One of Two Robots in URDF in Curobo?

### DISCUSSION THREAD

**Discussion by jwwang0-0** [NVlabs/curobo#427] [Software Q&A] [answered]
I'm using Curobo with two robots defined in a single URDF file. I’d like to disable collision checking for one of the robots under certain conditions to improve computational speed. I’m not concerned about the specific position or collision results for the inactive robot (I can set it to an initial location without obstacles), but I want to skip unnecessary collision calculations to boost simulation performance.

Is there a way to toggle collision checking on or off for the entire set of collision spheres of one robot at runtime, similar to how obstacle collision checks can be toggled? Is there a built-in command in Curobo to achieve this without needing to reload the environment?

Any advice or code examples for implementing this would be greatly appreciated!

**Comment by balakumar-s** [comment, marked as answer]
You can use this api to disable and enable collisions at runtime: https://curobo.org/_api/curobo.cuda_robot_model.types.html#curobo.cuda_robot_model.types.KinematicsTensorConfig.disable_link_spheres

From a motion_gen instance, you can acess it as ``motion_gen.robot_cfg.kinematics.kinematics_config``

**Reply by jwwang0-0** [reply]
Thank you!  I will test it out!
